### PR TITLE
Build 'Add hardware' dropdown.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done
- Refactored `Machines.js`, basically just by extracting out `HeaderStrip` into a separate component.
- Built "Add hardware" dropdown in machine list.
- Added placeholder `AddMachine` and `AddChassis` form components.
- Changed how `ContextualMenu` calculates position - @huwshimi not sure why but before I changed it the dropdown would only render after opening the dropdown twice? Some sort of weird race condition with the `useEffects`.... anyway it seems to work now.
- Tidied up spacing for some Loaders.

## QA
- Check that there is an "Add hardware" dropdown in the top-right of the machine list.
- Check that it is not there in the pools tab.
- Check that you are redirected to the Add machine and Add chassis forms when you click the links.
- Check that going to `/machines/cyw` takes you to an error page

## Fixes
Fixes canonical-web-and-design/MAAS-squad#1728

## Screenshot
0 0 0 0_8400_MAAS_r_machines (7)

### CODE REVIEW

File: ui/src/app/machines/views/Machines.js [hunk 1]
```diff
@@ -1,72 +1,139 @@
...
+              </Row>
+            </>
+          }
+        >
+          <Route exact path="/machines" component={MachineList} />
+          <Route exact path="/pools" component={Pools} />
+        </Section>
+      </Route>
+      <Route exact path="/machines/add-machine" component={AddMachine} />
+      <Route exact path="/machines/add-chassis" component={AddChassis} />
```
Comment (squidsoup) [inline ui/src/app/machines/views/Machines.js]: How about these routes instead?

`/machines/add` and `/machines/chassis/add`

Comment (anthonydillon) [reply ui/src/app/machines/views/Machines.js]: Might be odd if /machines/chassis is not a destination.

Comment (anthonydillon) [reply ui/src/app/machines/views/Machines.js]: Might be odd if /machines/chassis is not a destination.

Comment (Caleb-Ellis) [reply ui/src/app/machines/views/Machines.js]: Maybe `/machines/add` and `/machines/add-chassis`? A chassis isn't really a *thing*, but rather a convenience for adding multiple VMs at once so I think it probably shouldn't get its own subpath.

File: ui/src/app/machines/components/HeaderStrip/HeaderStrip.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  const machinesLoaded = useSelector(machineSelectors.loaded);
+  const resourcePools = useSelector(resourcePoolSelectors.all);
+  const resourcePoolsLoaded = useSelector(resourcePoolSelectors.loaded);
+  const navigationOptions = useSelector(generalSelectors.navigationOptions.get);
+  const { location } = useRouter();
+
+  useEffect(() => {
+    dispatch(poolActions.fetch());
+    dispatch(machineActions.fetch());
+  }, [dispatch]);
```
Comment (squidsoup) [inline ui/src/app/machines/components/HeaderStrip/HeaderStrip.js]: Nice to have this in one place, pleased you refactored this, thanks!

**Comment by squidsoup** [comment]
@Caleb-Ellis merging this as it affects the branch I'm working on, hope you don't mind!

**Pull request merged**